src/steps/domains/index.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  createDirectRelationship,
+  Entity,
+  IntegrationMissingKeyError,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  RelationshipClass,
+} from '@jupiterone/integration-sdk-core';
+import { createAPIClient } from '../../client';
+import { IntegrationConfig } from '../../config';
+import { ACCOUNT_ENTITY_KEY } from '../account';
+import { Entities, Relationships, Steps } from '../constants';
+import { createDomainEntity } from './converter';
+
+export async function fetchDomains({
+  instance,
+  jobState,
+  logger,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const accountEntityKey = (await jobState.getData(
+    ACCOUNT_ENTITY_KEY,
+  )) as string;
+  const accountEntity = (await jobState.findEntity(accountEntityKey)) as Entity;
+
+  if (!accountEntity) {
+    throw new IntegrationMissingKeyError(
+      `Expected to find Account entity in jobState.`,
+    );
+  }
+  const apiClient = createAPIClient(instance.config, logger);
+  const domains = await apiClient.getDomains();
+
+  for (const domain of domains) {
```

Comment:
Any particular reason you opted to use for loop after grabbing all the data rather than iterator?

This is the typical pattern we use for this: 
```ts
await apiClient.iterateDomains(async (domain) => {
  // Do something
  const domainEntity = createDomainEntity(domain);
  await jobState.addEntity(domainEntity);
 }
```
Where the client function signature looks like this:

```ts
  public async iterateDomains(
    iteratee: ResourceIteratee<Device>,
  ): Promise<void> {
```

Reply:
I personally didn't feel that it made the code any more readable, but I am happy to stick with this pattern. You mentioned breaking up `client.ts` a bit, and I plan on doing so on the third PR I plan on making to this repo which will introduce pagination support for iterating users and training participants. I will likely end up either adopting this pattern to make the surrounding pagination logic more readable, or discuss an alternative with you.